api/queries_sponsors_test.go

```diff
@@ -0,0 +1,118 @@
+package api
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/cli/cli/v2/pkg/httpmock"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_querySponsors(t *testing.T) {
+	var tests = []struct {
+		name           string
+		httpStubs      func(*httpmock.Registry)
+		expectedResult SponsorQuery
+		expectError    bool
+	}{
+		{
+			name: "success",
+			httpStubs: func(reg *httpmock.Registry) {
+				reg.Register(
+					httpmock.GraphQL(`query SponsorsList`),
```

Comment:
You've done a great job copying the existing patterns here. I would argue that the testing harness here is not sufficient. For example, if we change query variables to:

```
	variables := map[string]interface{}{
		"username": "lolnope",
	}
```

No test fails.

If we change the query to:

```
	query := `query SponsorsList($username: String!) {

	}`
```

No test fails.

---

It's possible to get away with this kind of missing behaviour because for the most part it's _really_ obvious when you do or don't have the data you expect in manual testing. Personally, I don't think that we should be content with this though 😬

Reply:
More on other possibilities at another time, I'm just whetting your appetite about some sketchy stuff.

Reply:
Yeah, I agree. This test suite is in no-way complete. There's been a certain tension between my preferred TDD loop and trying to adopt existing patterns. I've been focusing on understanding the code and trying to understand the patterns, but do plan to make this more robust. Thanks for the callout

Reply:
Ah sorry, my message was not clear at all. I meant the http registry test harness was lacking and the tests you copied from weren't covering that. Wasn't a criticism of your approach. 

Though when i think about it there may be a couple of helpers i forgot about. Need to check another time.

Reply:
Ah, for sure. In that regard, I've been trying to limit myself to adding pieces to the test harness as I discover a need for them. That way I'm being intentional about what's getting added. I think I'll continue this approach for now and 🤞 that the need for the missing pieces becomes clear

Reply:
Update: nothing else has illuminated itself through development, so now it's time to ponder this 🤔

Reply:
I think I've addressed this if you can take another look

Reply:
Definitely an improvement. What happens if you change the query in the implementation to:

```
	query := `query SponsorsList($username: String!) {

	}`
```

?

Personally, I think that most of the tests in the CLI that take the form:

```
					httpmock.GraphQL(`query SponsorsList`),
```

Are perhaps not _quite_ what I would hope because they don't validate the query shape. What do you think?